File: src/iindex.c
#include <R.h>
void iindex(int *iindex,
	    int *strata,
	    double *L,
	    double *R,
	    double *U,
	    int *N,
	    int *NS){
  int s, i, k;
  k=0;
  for (s=0;s<(*NS-1);s++){
    i=0;
    for (i=0; i<*N;i++){
      if ((L[i]==R[i] && L[i]==U[s+1]) /* exact obs */
	  ||
	  (L[i]<U[s+1] && R[i]>U[s])){   /* [U[s],U[s+1]] intersects [L[i],R[i]] */
	iindex[k] = i+1;
	k++;
      }
    }
    strata[s]=k;
  }
}

File: R/iindex.R
iindex <- function (L,R,grid) {
  stopifnot((length(grid)>0)
            & (length(L)>0)
            & (length(R)>0))
  stopifnot(is.numeric(c(L,R,grid)))
  N <- length(L)
  NS <- length(grid)
  ind <- .C("iindexSRC",
            index = integer(N*NS),
            strata = integer(NS-1),
            as.double(L),
            as.double(R),
            as.double(grid),
            as.integer(N),
            as.integer(NS),
            PACKAGE="prodlim")
  strata <- ind$strata
  index <- ind$index[1:max(strata)]
  list(iindex=index,imax=strata)
}
